### ISSUE TITLE
transform-runtime transpiles non-CallExpressions of Object.assign

### ISSUE THREAD

**Issue by eps1lon** [babel/babel#10859]
## Bug Report

**Current Behavior**
`Object.assign` will be transpiled by transform-runtime

**Input Code**
- REPL or Repo link if applicable: https://github.com/babel/babel/blob/3303b079c596e56909adff4b834f098f5f1d8e2a/packages/babel-plugin-transform-runtime/test/fixtures/runtime-corejs3/built-in-static-methods-proposals/input.js#L51

```js
Object.assign
```

**Expected behavior/code**
A read of a function should not cause transpilation since this might when running feature detection. It should match how transform-object-assign behaves.

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
See test file

**Environment**
babel repo

**Possible Solution**
Provide transform-runtime with information which static properties are functions. `transform-object-assign` isn't transpiling non-CallExpressions either.

**Comment by zloirock** [comment]
`var assign = Object.assign` is also non-`CallExpressions`. If you write something like in your example - it's just a problem of your code.

**Comment by nicolo-ribaudo** [comment]
Yes, Babel should simulate a native environment as much as possible. It can (and does) do much more than simply replacing function calls.

**Comment by eps1lon** [comment]
> var assign = Object.assign is also non-CallExpression

I'm not sure how you got the impression that I think otherwise?

>  If you write something like in your example above - it's just a problem of your code.

`if (Object.assign === undefined)` is a problem of my code, why?

> Yes, Babel should simulate a native environment as much as possible. 

Just to clarify: Would you agree with me that the current behavior is a problem or not?

>  it should be transpiled since it's a part of features detection and it's available

These statements contradict each other. If I'm doing feature detection it shouldn't be transpiled. Otherwise the feature detection is useless: If `if (Object.assign === undefined)` is transpiled then it will be equivalent to `if (false)` and is subject to dead code elimination.

Could any of you adress the behavior mismatch between transform-runtime and transform-object-assign?

**Comment by zloirock** [comment]
A simple example:
```js
if (!Object.assign) throw TypeError('Please, use engines with `Object.assign` support');
```

**Comment by eps1lon** [comment]
> A simple example:

For what? 

We don't want to throw in the block. Just skip using e.g. logging libraries that need Object.assign.

**Comment by zloirock** [comment]
The example that if it will not be transpiled users will have an error instead of returning work like in engines with native `Object.assign` support.

**Comment by eps1lon** [comment]
If that is a concern then don't change it. But you commented on the linked PR introducing even more transpilation. At least leave the status quo so that the behavior stays the same.

Third times the charm: Why does transform-object-assign not work this way?

**Comment by eps1lon** [comment]
Also 
> if (!Object.assign) throw TypeError('Please, use engines with `Object.assign` support');

This is still subject to dead code elimination. Why would you use this code and the transform?

**Comment by zloirock** [comment]
`transform-object-assign` is not my work, so I have no ideas.

> This is still subject to dead code elimination.

Yes, it's the case for dead code elimination and I think that this transform should be improved in the future. But cases could be more complex and dead code elimination will not work.
